Code/Legacy/CrySystem/LevelSystem/SpawnableLevelSystem.cpp

```diff
@@ -235,6 +235,16 @@ namespace LegacyLevelSystem
             return false;
         }
 
+        // This is a valid level, find out if any systems need to stop level loading before proceeding
+        for (const auto& listener : m_listeners)
+        {
+            if (listener->BlockLoading(validLevelName.c_str()))
+            {
+                AZ_TracePrintf("CrySystem::SpawnableLevelSystem", "LoadLevel for %s was blocked.\n", validLevelName.c_str());
+                return false;
+            }
+        }
```

Comment:
Instead of adding a dependency on the `ILevelSystemListener`, what could be done is to add an EBus interface(or an AZ::Event object, to a class like AzFramework::Application or AZ::IO::IArchive) to AzFramework, and then implement it here.
It can sends out the ShouldBlockLoading event.

You can see how the Level System handlers an AZ::Event from the Archive system that signals when the level file has been [opened or closed](https://github.com/o3de/o3de/blob/development/Code/Legacy/CrySystem/LevelSystem/LevelSystem.h#L158-L159).

Reply:
Thank you! Added a bus to AzFramework instead of AZ::Event since I dont think Events were meant to return a result. Let me know what you think, now I want to update LyShine and Audio to also use the AzFramework bus so they dont need CryLegacy Level stuff, that could be another PR :P